hypothesis-python/tests/numpy/test_gen_data.py

```diff
@@ -1112,18 +1112,22 @@ def test_basic_indices_replaces_whole_axis_slices_with_ellipsis(idx):
     assert slice(None) not in idx
 
 
-@given(
-    shape=nps.array_shapes(min_dims=0, max_side=4)
-    | nps.array_shapes(min_dims=0, min_side=0, max_side=10),
-    min_dims=st.integers(0, 5),
-    allow_ellipsis=st.booleans(),
-    allow_newaxis=st.booleans(),
-    data=st.data(),
-)
-def test_basic_indices_generate_valid_indexers(
-    shape, min_dims, allow_ellipsis, allow_newaxis, data
-):
-    max_dims = data.draw(st.none() | st.integers(min_dims, 32), label="max_dims")
+@given(st.data())
+def test_basic_indices_generate_valid_indexers(data):
```

Comment:
I'd like to keep the independent parameters (everything except `max_dims`) as arguments supplied by `@given` - it's just a little cleaner and the reporting is easier to read if it fails.  In this case the diff will also be smaller.

Reply:
Done, makes sense now. Note that `min_dims` also needs to be drawn in the test body so that it's `< len(shape)` when `allow_newaxis == False` (which I believe should also be deprecated behaviour, see comment). I've also reflected this style in the equivalent test for `extra.array_api`.